### PR TITLE
Relocate AMP Preview Button

### LINKED ISSUE #7896: Add AMP under "Preview" dropdown in WordPress
Hi! I wanted to let you know that it's now possible to add items under "Preview" menu in Gutenberg since: 
- https://github.com/WordPress/gutenberg/pull/64644

More here: https://make.wordpress.org/core/2024/10/18/extending-the-preview-dropdown-menu-in-wordpress-6-7/

I recommend moving the AMP preview there from the plugin toolbar conditionally when `PreviewDropdownMenuItem` is available:

Screenshot 2024-10-17 at 20 22 59

The original issue at Gutenberg repo used AMP as an example of why the Preview dropdown had to be extendable: 
- https://github.com/WordPress/gutenberg/issues/25309

I wrote more ideas here on how to improve preview extensibility. Feel free to share AMP perspective!

- https://github.com/WordPress/gutenberg/issues/65005

### PR DESCRIPTION
**Pull request by Takshil-Kunadia**
## Summary

- Fixes #7896 
- AMP under "Preview" dropdown using the `PluginPreviewMenuItem`.

## Checklist

- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: assets/src/block-editor/components/preview-menu-item.js [hunk 1]
```diff
@@ -221,37 +220,36 @@ class AmpPreviewButton extends Component {
 			isEnabled &&
 			!errorMessages.length &&
 			!isStandardMode && (
-				<Button
+				<PluginPreviewMenuItem
```
Comment (westonruter) [inline assets/src/block-editor/components/preview-menu-item.js:227]: We'll need to make sure that if `PluginPreviewMenuItem` doesn't exist due to not being on the latest version of WordPress that we short-circuit this function so we don't get an error.

File: bin/local-env/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM wordpress
+# @todo: switch to wordpress:latest once it's available
+FROM wordpress:beta-6.7-php8.3-apache
```
Comment (thelovekesh) [inline bin/local-env/Dockerfile:2]: Required to test AMP preview button changes.

**Comment by thelovekesh** [comment]
JS Unit tests are (not) failing because codecov is unable to upload reports of the PR from fork.

**Comment by westonruter** [comment]
I noticed that the button of the AMP icon is ever so slightly cut off:

image

File: assets/src/block-editor/plugins/amp-preview-item.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Internal dependencies
+ */
+import { AMPFilledIcon } from '../../icons';
+import AmpPreviewMenuItem from '../components/preview-menu-item';
+
+export const name = 'amp-preview-menu-item';
+
+export const icon = (
+	<AMPFilledIcon viewBox="0 0 62 62" height={18} width={18} />
```
Comment (westonruter) [inline assets/src/block-editor/plugins/amp-preview-item.js:10]: The styling of the menu items is a bit different:

image

image

Maybe this is contributing to the icon being cropped at the bottom.

Comment (westonruter) [reply assets/src/block-editor/plugins/amp-preview-item.js:10]: If I make the icon have the 24x24 pixel dimensions then it isn't cropped. But then it's probably too big.

**Comment by westonruter** [comment]
As expected, in 6.6.x I'm not seeing any preview button:

image

And when updating to 6.7-RC5 I see the button as expected:

image

**Comment by westonruter** [comment]
This seems to work well. The only outstanding item is the slight cropping on the icon.

File: assets/src/block-editor/plugins/amp-preview-item.js (hunk 3 above)
Comment (thelovekesh) [reply assets/src/block-editor/plugins/amp-preview-item.js:10]: Interesting, With the latest build on this PR, it looks like

image

Comment (Takshil-Kunadia) [reply assets/src/block-editor/plugins/amp-preview-item.js:10]: Likewise @thelovekesh, The latest build doesn't have the cropping issue on any breakpoints.

Screenshot 2024-11-12 at 3 11 22 PM

Comment (westonruter) [reply assets/src/block-editor/plugins/amp-preview-item.js:10]: Hummm. I still see it cropped with after doing a fresh `npm run build:prod`:

image

If I increase my browser's zoom level from 100% to 110% then the issue goes away:

image

Maybe it's a ChromeOS-specific rendering issue? In any case, not critical. Would still be good to debug further.

**Review by westonruter — APPROVED** [review]

**Pull request merged**